vicodyn/src/vicodyn/queue/invocation.cpp

```diff
@@ -37,28 +39,25 @@ auto invocation_t::absorb(invocation_t& other) -> void {
     });
 }
 
-auto invocation_t::append(const msgpack::object& message,
-                          uint64_t event_id,
-                          hpack::header_storage_t headers,
-                          const io::graph_node_t& incoming_protocol,
-                          stream_ptr_t backward_stream) -> std::shared_ptr<stream_t>
-{
+auto invocation_t::append(invocation_request_t invocation) -> std::shared_ptr<stream_t> {
     auto forward_stream = std::make_shared<stream_t>(stream_t::direction_t::forward);
 
     operation_t operation = operation_t();
-    operation.event_id = event_id;
-    operation.headers = std::move(headers);
+    operation.event_id = invocation.message().type();
+    operation.headers = std::move(invocation.message().headers());
```

Comment:
Haven't dive too deep, but is is always safe to move headers out of referenced message?